nifi-commons/nifi-record/src/test/java/org/apache/nifi/serialization/record/field/TestObjectLocalDateTimeFieldConverter.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.serialization.record.field;
+
+import org.junit.jupiter.api.Test;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class TestObjectLocalDateTimeFieldConverter {
+
+    @Test
+    public void testConvertTimestampMillis() {
+        final ObjectLocalDateTimeFieldConverter converter = new ObjectLocalDateTimeFieldConverter();
+        final long now = System.currentTimeMillis();
+        final Instant instant = Instant.ofEpochMilli(now);
+        final LocalDateTime expected = LocalDateTime.ofInstant(instant, ZoneId.systemDefault());
+
+        final LocalDateTime result = converter.convertField(now, Optional.empty(), "test");
+        assertEquals(expected, result);
+        assertEquals(now, result.atZone(ZoneId.systemDefault()).toInstant().toEpochMilli());
```

Comment:
Is there a particular reason for this comparison after comparing the expected value above? Converting to a different representation is effectively mutating the value, versus testing the `LocalDateTime` conversion on its own.